File: java/src/test/java/challenges/aoc2024/day1/model/ListLinePairTest.java
package challenges.aoc2024.day1.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class ListLinePairTest {

    @Test
    void shouldBuildListLinePairObjectFromLeftAndRightValues() {
        // When
        final ListLinePair listLinePair = ListLinePair.of(23, 47);

        // Then
        assertEquals(ImmutableListLinePair.class, listLinePair.getClass());
        assertEquals(23, listLinePair.getLeftValue());
        assertEquals(47, listLinePair.getRightValue());
    }
}

File: scala/src/test/scala/challenges/aoc2024/day1/model/ListLinePairTest.scala
package challenges.aoc2024.day1.model

import org.scalatest.funsuite.AnyFunSuite

class ListLinePairTest extends AnyFunSuite {

  test("should build ListLinePair object from left and right values") {
    // When
    val listLinePair: ListLinePair = ListLinePair.of(23, 47)

    // Then
    assert(listLinePair.isInstanceOf[ListLinePair])
    assert(23 == listLinePair.leftValue)
    assert(47 == listLinePair.rightValue)
  }
}
